src/client/datascience/notebook/vscodeNotebookController.ts

```diff
@@ -112,6 +112,19 @@ export class VSCodeNotebookController implements Disposable {
         }
     }
 
+    // Handle the execution of notebook cell
+    public async handleExecution(cells: NotebookCell[]) {
```

Comment:
Previously I used to hide something like this behind a test interface. Honestly I'm not sure that it matters? Much of the work of kernel push was so that notebookControllers are not exposed outside of their extensions. I just made it public so that tests can execute, but I can hide it if needed.